transactionverifier/transactionVerifier.go

```diff
@@ -0,0 +1,211 @@
+package transactionverifier
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/stateproofs/transactionverificationtypes"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+var (
+	ErrUnsupportedHashFunction      = errors.New("proof hash function is unsupported")
+	ErrProofLengthTreeDepthMismatch = errors.New("proof length and tree depth do not match")
+	ErrRootMismatch                 = errors.New("root mismatch")
+	ErrInvalidTreeDepth             = errors.New("invalid tree depth")
+	ErrIndexDepthMismatch           = errors.New("node index is not smaller than 2^depth")
+	ErrInvalidPosition              = errors.New("invalid position for node")
+)
+
+type NodePosition int
+
+const (
+	leftChild NodePosition = iota
+	rightChild
+)
+
+// computeTransactionLeaf receives the transaction ID and the signed transaction in block's hash, and computes
+// the leaf of the vector commitment associated with the transaction.
+// Parameters:
+// txId - the Sha256 hash of the msgpacked transaction.
+// stibHash - the Sha256 of the msgpacked transaction as it's saved in the block.
+func computeTransactionLeaf(txId types.Digest, stibHash types.Digest) types.Digest {
+	leafDomainSeparator := []byte(transactionverificationtypes.TxnMerkleLeaf)
+	leafData := leafDomainSeparator
+	leafData = append(leafData, txId[:]...)
+	leafData = append(leafData, stibHash[:]...)
+
+	// The leaf returned is of the form: Sha256("TL" || Sha256(transaction) || Sha256(transaction in block))
+	return sha256.Sum256(leafData)
+}
+
+// computeLightBlockHeaderLeaf receives the parameters comprising a light block header, and computes the leaf
+// of the vector commitment associated with the transaction.
+// Parameters:
+// roundNumber - the round of the block to which the light block header belongs.
+// transactionCommitment - the sha256 vector commitment root for the transactions in the block to which the light block header belongs.
+// genesisHash - the hash of the genesis block.
+// seed - the sortition seed of the block associated with the light block header.
```

Comment:
I wish we had some description on what the seed is and where to get it.

Reply:
It is, the light block header is a subset of some the fields of the normal block header, the seed being among them. Where do you think this should be explained?

Reply:
I think the current comment is sufficient